From 80afd20c19389ffbae7a05f0e197dd1db50289ad Mon Sep 17 00:00:00 2001
From: Niels De Graef <nielsdegraef@gmail.com>
Date: Sat, 19 Jan 2019 11:57:22 +0100
Subject: [PATCH] Add support for g_autoptr() to our types

g_autoptr() is a macro that was defined in GLib 2.44 that allows for
basic auto-cleanup of variables. One way to add this kind of support
would be through the use of e.g. `G_DECLARE_DERIVABLE_TYPE()` for our
declarations, but this would consitute an ABI break (due to the
`...Private *` field in the public structs). Instead, we can use
`G_DEFINE_AUTOPTR_CLEANUP_FUNC` to manually declare this.

This commit also bumps the minimally required GLib version to 2.44
---
 configure.ac                  | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index 09080e0..b1e6bcc 100644
--- a/configure.ac
+++ b/configure.ac
@@ -9,9 +9,9 @@ dnl Dependency versions
 
 # Required
 
-GLIB_REQ=2.38.0
-GLIB_MIN=GLIB_VERSION_2_38
-GLIB_MAX=GLIB_VERSION_2_38
+GLIB_REQ=2.44.0
+GLIB_MIN=GLIB_VERSION_2_44
+GLIB_MAX=GLIB_VERSION_2_44
 
 # Optional
 
-- 
2.18.1

